library/x509.c

```diff
@@ -805,6 +805,75 @@ static char nibble_to_hex_digit(int i)
     return (i < 10) ? (i + '0') : (i - 10 + 'A');
 }
 
+/* Return the x.y.z.... style numeric string for the given OID */
```

Comment:
Shouldn't this also be guarded by `#if defined(MBEDTLS_OID_C)`?

Reply:
The functions are moving from the OID module to an X.509 module. So none of the code added by this PR should be guarded by `MBEDTLS_OID_C`. Instead it should be guarded by whatever `MBEDTLS_X509_XXX` options are needed at the point where the X.509 code uses them.